--- a/arch/arm/mach-msm/include/mach/irqs.h
+++ b/arch/arm/mach-msm/include/mach/irqs.h
@@ -97,11 +97,10 @@
 #define NR_GPIO_IRQS 117
 #define NR_QPNP_IRQS 32768
 #define NR_BOARD_IRQS NR_QPNP_IRQS
-#elif defined(CONFIG_ARCH_MSM9625)
+#else
 #define NR_MSM_IRQS 288
 #define NR_GPIO_IRQS 76
 #define NR_BOARD_IRQS 0
-#endif
+#endif
 #define NR_IRQS (NR_MSM_IRQS + NR_GPIO_IRQS + NR_BOARD_IRQS)
 #define MSM_GPIO_TO_INT(n) (NR_MSM_IRQS + (n))
 #define FIRST_GPIO_IRQ MSM_GPIO_TO_INT(0)