--- a/arch/arm/mm/proc-v7.S
+++ b/arch/arm/mm/proc-v7.S
@@ -89,7 +89,11 @@ ENDPROC(cpu_v7_dcache_clean_area)
 	.type	cpu_v7_name, #object
 cpu_v7_name:
 	.arm
-	string cpu_v7_name,"ARMv7 Processor"
+	.ascii	"ARMv7 Processor"
+	.size	cpu_v7_name, . - cpu_v7_name
+	.type	cpu_arch_name, #object
+cpu_arch_name:
+	.ascii	"armv7"
 	.size	cpu_arch_name, . - cpu_arch_name
 
 	/*
@@ -339,8 +343,10 @@ __v7_setup_stack:
 
 	__CPUINIT
 
-	string cpu_arch_name,"armv7"
-	string cpu_elf_name,"v7"
+	.type	cpu_elf_name, #object
+cpu_elf_name:
+	.ascii	"v7"
+	.size	cpu_elf_name, . - cpu_elf_name
 	.align
 
 	.section ".proc.info.init", #alloc, #execinstr
