--- a/arch/arm/kernel/asm-offsets.c
+++ b/arch/arm/kernel/asm-offsets.c
@@ -8,11 +8,16 @@
  * it under the terms of the GNU General Public License version 2 as
  * published by the Free Software Foundation.
  */
+#include <linux/types.h>
+#include <linux/stddef.h>
+#include <linux/mm_types.h>
+#include <linux/page-constants.h>
 #include <linux/sched.h>
 #include <linux/mm.h>
 #include <linux/dma-mapping.h>
-#include <asm/thread_info.h>
+#include <linux/string.h>
 #include <asm/memory.h>
+#include <asm/thread_info.h>
 #include <asm/procinfo.h>
 
 int main(void)