--- a/arch/arm/kernel/entry-armv.S
+++ b/arch/arm/kernel/entry-armv.S
@@ -1013,10 +1013,10 @@ ENDPROC(__und_usr)
  */
 	.align	5
 __und_usr_thumb:
-	sub	r4, r2, #2			@ First half of thumb instruction at LR - 2
+	sub	r4, r2, #2
 	/*
 	 * Check whether the instruction is a 32-bit Thumb-2 instruction.
-	 * If r5 == 0, it is a 16-bit Thumb instruction.
+	 * If not, it is a 16-bit Thumb instruction.
 	 */
 #ifdef CONFIG_CPU_ENDIAN_BE8
 	ldrht	r5, [r4]
@@ -1034,10 +1034,10 @@ __und_usr_thumb:
  */
 	.align	5
 __und_usr_arm:
-	sub	r4, r2, #4			@ ARM instr at LR - 4
+	sub	r4, r2, #4
 #ifdef CONFIG_CPU_ENDIAN_BE8
 	rev	r0, r0				@ little endian instruction
 #endif
-	str	r0, [sp, #S_PC]			@ save for later
+	str	r0, [sp, #S_PC]
 	b	call_undef
 ENDPROC(__und_usr_arm)
 
@@ -1055,10 +1055,10 @@ ENDPROC(__und_usr_arm)
 	.align	5
 __pabt_usr:
 	sub	sp, sp, #S_FRAME_SIZE
-	stmia	sp, {r0 - r12}			@ Save r0 - r12
-	ldmia	r1, {r3 - r5}			@ Get USR pc, cpsr
-	stmdb	r0, {sp, lr}^			@ Save sp_usr, lr_usr
-	clrex					@ Clear exclusive monitor
+	stmia	sp, {r0 - r12}
+	ldmia	r1, {r3 - r5}
+	stmdb	r0, {sp, lr}^
+	clrex
 	@
 	@ Call main routine
 	@
@@ -1076,10 +1076,10 @@ ENDPROC(__pabt_usr)
 	.align	5
 __dabt_usr:
 	sub	sp, sp, #S_FRAME_SIZE
-	stmia	sp, {r0 - r12}			@ Save r0 - r12
-	ldmia	r1, {r3 - r5}			@ Get USR pc, cpsr
-	stmdb	r0, {sp, lr}^			@ Save sp_usr, lr_usr
-	clrex					@ Clear exclusive monitor
+	stmia	sp, {r0 - r12}
+	ldmia	r1, {r3 - r5}
+	stmdb	r0, {sp, lr}^
+	clrex
 	@
 	@ Call main routine
 	@
