--- a/arch/arm/kernel/entry-armv.S
+++ b/arch/arm/kernel/entry-armv.S
@@ -1176,7 +1176,7 @@ ENDPROC(__und_usr)
 	.align	5
 __pabt_usr:
 	usr_entry
-	mov	r2, sp				@ regs
+	mov	r2, sp
 	pabt_helper
  UNWIND(.fnend		)
 	/* fall through */
@@ -1200,7 +1200,7 @@ ENDPROC(__pabt_usr)
 	.align	5
 __dabt_usr:
 	usr_entry
-	mov	r2, sp				@ regs
+	mov	r2, sp
 	dabt_helper
 	b	ret_from_exception
  UNWIND(.fnend		)
@@ -1224,7 +1224,7 @@ ENTRY(ret_from_exception)
 ENDPROC(__dabt_usr)
 ENDPROC(ret_from_exception)
 
-	.align	5
+	.align	2
 __irq_usr:
 	usr_entry
 	kuser_cmpxchg_check
@@ -1264,7 +1264,7 @@ ENDPROC(__irq_svc)
 	.popsection
 #endif
 
-	.align	5
+	.align	2
 __und_svc:
 #ifdef CONFIG_KPROBES
 	@ If a kprobe is about to simulate a "stmdb sp!, {..., pc}" instruction,
