cat > 61-fix-cache-v7-macros.patch << 'EOF'
--- a/arch/arm/mm/cache-v7.S
+++ b/arch/arm/mm/cache-v7.S
@@ -91,11 +91,11 @@
 	and	r1, r1, #7		@ mask of the bits for current cache only
 	cmp	r1, #2			@ see what cache we have at this level
 	blt	skip			@ skip if no cache, or just i-cache
- ARM(	orr	r11, r10, r9, lsl r5	)	@ factor way and cache number into r11
- THUMB(	lsl	r6, r9, r5		)
- THUMB(	orr	r11, r10, r6		)	@ factor way and cache number into r11
- ARM(	orr	r11, r11, r7, lsl r2	)	@ factor index number into r11
- THUMB(	lsl	r6, r7, r2		)
- THUMB(	orr	r11, r11, r6		)	@ factor index number into r11
+	orr	r11, r10, r9, lsl r5
+	nop
+	nop
+	orr	r11, r11, r7, lsl r2
+	nop
+	nop
 	mcr	p15, 0, r11, c7, c14, 2		@ clean & invalidate by set/way
 	subs	r7, r7, #1			@ decrement the index
 	bge	loop2
@@ -126,8 +126,8 @@
  *	- end     - virtual end address of region
  */
 ENTRY(v7_flush_dcache_all)
- ARM(	stmfd	sp!, {r4-r5, r7, r9-r11, lr}	)
- THUMB(	stmfd	sp!, {r4-r7, r9-r11, lr}	)
+	stmfd	sp!, {r4-r5, r7, r9-r11, lr}
+	nop
 	dmb					@ ensure ordering with previous memory accesses
 	mrc	p15, 1, r0, c0, c0, 1		@ read clidr
 	ands	r3, r0, #0x7000000		@ extract loc from clidr
@@ -132,8 +132,8 @@
 	ands	r3, r0, #0x7000000		@ extract loc from clidr
 	mov	r3, r3, lsr #23			@ left align loc bit field
 	beq	finished			@ if loc is 0, then no need to clean
- ARM(	ldmfd	sp!, {r4-r5, r7, r9-r11, lr}	)
- THUMB(	ldmfd	sp!, {r4-r7, r9-r11, lr}	)
+	ldmfd	sp!, {r4-r5, r7, r9-r11, lr}
+	nop
 	dsb					@ WFI may enter a low-power mode
 	mov	pc, lr
 ENDPROC(v7_flush_kern_cache_all)
@@ -144,8 +144,8 @@
  *	- end     - virtual end address of region
  */
 ENTRY(v7_flush_dcache_louis)
- ARM(	stmfd	sp!, {r4-r5, r7, r9-r11, lr}	)
- THUMB(	stmfd	sp!, {r4-r7, r9-r11, lr}	)
+	stmfd	sp!, {r4-r5, r7, r9-r11, lr}
+	nop
 	dmb					@ ensure ordering with previous memory accesses
 	mrc	p15, 1, r0, c0, c0, 1		@ read clidr, r0 = clidr
 	ands	r3, r0, #0xe00000		@ extract LoUIS from clidr
@@ -150,8 +150,8 @@
 	ands	r3, r0, #0xe00000		@ extract LoUIS from clidr
 	mov	r3, r3, lsr #20			@ r3 = LoUIS * 2
 	beq	finished			@ if LoUIS is 0, then no need to clean
- ARM(	ldmfd	sp!, {r4-r5, r7, r9-r11, lr}	)
- THUMB(	ldmfd	sp!, {r4-r7, r9-r11, lr}	)
+	ldmfd	sp!, {r4-r5, r7, r9-r11, lr}
+	nop
 	dsb					@ WFI may enter a low-power mode
 	mov	pc, lr
 ENDPROC(v7_flush_dcache_louis)
EOF
