--- a/arch/arm/vfp/entry.S
+++ b/arch/arm/vfp/entry.S
@@ -27,9 +27,9 @@
 	@ Save the current VFP state
 	@ r0 = VFP workspace base
 	@ r1 = thread_info base
-	str	r2, [sp, @ S_PC]		@ update regs->ARM_pc for Thumb 2 case
+	str	r2, [sp, #S_PC]		@ update regs->ARM_pc for Thumb 2 case
 	ldr	r4, .LCvfp
-	ldr	r11, [r10, @ TI_CPU]	@ CPU number
-	add	r10, r10, @ TI_VFPSTATE	@ r10 = workspace
+	ldr	r11, [r10, #TI_CPU]	@ CPU number
+	add	r10, r10, #TI_VFPSTATE	@ r10 = workspace
 	
 	ldr	pc, [r12, r11, lsl #2]	@ call the emulation routine
